plugins/offboard/offboard.h

```diff
@@ -152,6 +158,13 @@ class Offboard : public PluginBase
      */
     void set_velocity_body(VelocityBodyYawspeed velocity_body_yawspeed);
 
+    /**
+     * @brief Set the actuator control targets for a specific control group.
+     *
+     * @param actuator_control_target float[8] and control_group uint8_t.
+     */
+    void set_actuator_control_target(ActuatorControlTarget actuator_control_target, ActuatorControlGroup control_group);
```

Comment:
The name `set_actuator_control_target()` sounds to me like you're setting what target you want to affect, and there would be another control that sets the value of that target. However I'm fairly sure that this actually sets the value of the target - correct? Perhaps `set_actuator_control`  would actually be more clear?

Reply:
I vote to use different naming from mavlink for the API here and name it what it actually does.

Also, I don't think it's a good idea to have a `float[8]` but rather individual fields that all say what they are.

Basically, you have to imagine an app developer that doesn't know how the drone works and wants to command it without knowing what an actuator or an actuator group is.

Reply:
@julianoes I very much agree with you in principle, but I don't know whether this is feasible/reasonable.

So what has been proposed is a generic API for setting any target value for any [control group](https://dev.px4.io/en/concept/mixing.html#control-groups). There are 6 groups and they represent different uses of a physical bus of 8 outputs - e.g. flight control, payload, gimbal, manual passthrough. 

Choices are either to:
* Stick with current API - most generic approach
* Create (up to  6) separate methods that abstract the control groups. Then we can have specific names and types for each case
* Decide this API is too low level for our audience/core API and suggest this be provided as an external plugin by anyone who wants it. 

My problem here is that I don't really understand what, how and when each of the control groups would really be used - which makes it hard to assess whether we "really" need all or any of these. For example, I'm guessing group 0 is for any vehicle flight control and 1 is for a VTOL in MC mode, but I have no idea if that is correct, or what the system would do if you sent an invalid control group message for the current case, or why you would want to.

@julianoes Do you understand how the control groups would actually work?

@xgerrmann What is the use case you'd actually be trying to solve if you had the power of this API? Is it (say) to just drop a payload? If so, then it may be that we could just expose the payload control group.

Reply:
In my opinion if we just replicate the hard to understand somewhat generic mavlink interface, then DroneCore ends up being nothing else than what we already have with mavlink. It will expose a bunch of fields and no one knows what is supported and with what values/limits.

Therefore, we really need to break it down into what we're trying to achieve so that it is very clear to a future user of the API.

Therefore, I suggest to either change this one to something that is understandable and properly supported, or not include it for now but only in a fork for now and then separate plugin later.

> Do you understand how the control groups would actually work?

I'd have to loop up if/how it all works.